lib/k10/dtb.ex

```diff
@@ -221,25 +227,25 @@ defmodule K10.DTB do
 
   defp extract_structs(bin, strings) do
     # FIXME error handling
-    {<<>>, acc} = extract_structs(bin, strings, %{})
+    {<<>>, acc} = extract_structs(bin, strings, [])
     acc
   end
 
   defp extract_structs(binary, strings, acc) do
     case extract_struct(binary) do
       # root case
       {{:fdt_begin_node, ""}, rest} ->
-        {rest, acc} = extract_structs(rest, strings, %{})
+        {rest, acc} = extract_structs(rest, strings, [])
         {:continue, rest, acc}
 
       {{:fdt_begin_node, name}, rest} ->
-        {rest, nested} = extract_structs(rest, strings, %{})
-        {:continue, rest, Map.put_new(acc, name, nested)}
+        {rest, nested} = extract_structs(rest, strings, [])
+        {:continue, rest, List.keystore(acc, name, 0, {name, nested})}
```

Comment:
`Map.put_new` was only storing the first key-value pair if there're duplicates whereas this will store the last, right? Could that matter? Should we document it somewhere since it could be a little surprising (especially since it'll seem like it's out of order)

Reply:
I think this is not super relevant tbh. A valid FDT should not have repeated nodes/properties. We could be extra careful and check before we insert.